Tools/WebKitTestRunner/TestController.cpp

```diff
@@ -2714,6 +2731,12 @@ void TestController::decidePolicyForGeolocationPermissionRequestIfPossible()
             WKGeolocationPermissionRequestDeny(permissionRequest);
     }
     m_geolocationPermissionRequests.clear();
+
+    if (!permissionChanged)
+        return;
+
+    for (auto& originString : m_geolocationPermissionQueriers)
+        WKPagePermissionChanged(toWK("geolocation").get(), originString.get());
```

Comment:
Why not invoke `WKPagePermissionChanged` in `TestController::setGeolocationPermission`?
It's bit weird that the permission request handler takes a parameter called `permissionChanged`.

Reply:
Yeah, I initially did this because of a misinterpretation of the code. I wanted `WKPagePermissionChanged` to be invoked after the permission state gets changed locally. But that happens in `TestController::setGeolocationPermission`. So we can just invoke it after that. I'll fix this.